test/http/detail/chunk_encode.cpp

```diff
@@ -0,0 +1,132 @@
+//
+// Copyright (c) 2013-2016 Vinnie Falco (vinnie dot falco at gmail dot com)
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+
+#include <beast/http/detail/chunk_encode.hpp>
+#include <beast/core/to_string.hpp>
+#include <beast/unit_test/suite.hpp>
+
+namespace beast {
+namespace http {
+namespace detail {
+
+class chunk_encode_test : public beast::unit_test::suite
+{
+public:
+    struct final_chunk
```

Comment:
I don't think we need the `final_chunk` class. We could add an overload to `encode` that will let us know when we've reached the last arg. Commit showing what I have in mind: https://github.com/seelabs/Beast/commit/8d9db13782aa8655c71ac8c69daf2a5b096bc67d